admin/schema.php

```diff
@@ -16,44 +16,40 @@
 
 /**
  * MantisBT schema definition
- * The schema is defined a list of schema updates, stored as an array.
+ * The schema is defined as a numbered list of updates, stored as an array.
  *
  * Each upgrade step consists of two elements:
- * 1. The function to generate SQL statements. Available functions (from ADOdb
- *    library) are:
- *      CreateTableSQL, DropTableSQL, ChangeTableSQL, RenameTableSQL,
- *      RenameColumnSQL, DropTableSQL, ChangeTableSQL, RenameTableSQL,
- *      RenameColumnSQL, AlterColumnSQL, DropColumnSQL
- *    A local function "InsertData" has been provided to add data to the db
+ *
+ * 1. The function to generate SQL statements. Available functions are
+ *    - Data dictionary operations from ADOdb library; one of:
+ *      CreateTableSQL, ChangeTableSQL, RenameTableSQL, DropTableSQL,
+ *      AddColumnSQL, AlterColumnSQL, RenameColumnSQL, DropColumnSQL,
+ *      CreateIndexSQL, DropIndexSQL
+ *    - InsertData: local function to add data to the db
+ *    - UpdateFunction: local function to perform arbitrary changes;
+ *      the function must be defined in install_helper_functions_api.php
+ *      with 'install_' prefix
+ *    - null: no-op upgrade step
  *
  * 2. An array of the parameters to be passed to the function.
  *
  * The integrity of the schema relies on strict ordering of this array.
  * - ONLY ADD NEW CHANGES TO THE END OF THE TABLE!!!
+ *   Always specify the schema step (array key), for documentation purposes
  * - NEVER SKIP AN INDEX IN THE SEQUENCE!!!
+ *
+ * 'Release markers' are placed right AFTER the last schema step that is
+ * included in the corresponding release
  */
 
-if( !function_exists( 'db_null_date' ) ) {
-	/**
-	 * Legacy null date function for installer backwards compatibility
-	 * @return integer
-	 */
-	function db_null_date() {
-		return 0;
-	}
-}
-
 /**
- * Legacy date function for installer backwards compatibility
- * @return string
+ * Standard table creation options
+ * Array key is the ADOdb datadict driver's name
  */
-function installer_db_now() {
-	global $g_db;
-
-	# Timezone must not be set to UTC prior to calling BindTimestamp(), as
-	# ADOdb assumes a local timestamp and does the UTC conversion itself.
-	return $g_db->BindTimeStamp( time() );
-}
+$t_table_options = array(
```

Comment:
I think that $t_table_options for MySQL should include the default charset.
This is to cover the case where Mantis is installed in an existing database where the default charset is not utf8 or utf8mb4.